prow/git/v2/client_factory.go

```diff
@@ -339,9 +346,18 @@ func (c *clientFactory) ClientForWithRepoOpts(org, repo string, repoOpts RepoOpt
 		// something unexpected happened
 		return nil, err
 	} else {
-		// we have cloned the repo previously, but will refresh it
-		if err := cacheClientCacher.RemoteUpdate(); err != nil {
-			return nil, err
+		// We have cloned the repo previously, but will refresh it. By default
+		// we refresh all refs with a call to `git remote update`.
+		if repoOpts.FetchCommits == nil {
+			if err := cacheClientCacher.RemoteUpdate(); err != nil {
+				return nil, err
+			}
```

Comment:
The `NoFetchTags` option is ignored in this case. We don't actually need to fix this now so long as you document it in the field's GoDoc instead.